crates/rust-lib/src/lib.rs

```diff
@@ -572,8 +573,11 @@ pub trait RustGenerator<'a> {
         record: &Record,
         docs: &Docs,
         derive_component: bool,
+        additional_derives: Vec<String>,
```

Comment:
It's a bit unfortunate to pass this around in so many places. In theory this `rust-lib` crate shouldn't even exist any longer. This was only split out as it was shared between Wasmtime and a Rust guest, but Wasmtime no longer lives here nor uses this so it's an unnecessary abstraction nowadays. Additionally that's a large enough refactor there's no need to do it in this PR.

In the meantime though a possible alternative design would be a trait method on `RustGenerator` along the lines of `fn additional_derives(&self) -> &[String]`. I think that should avoid most of the passing around and clean that up slightly. Mind taking this on and I'll as a follow up work on merging this crate into the `rust` crate?

Reply:
Yep, I'll shift it around. Honestly was having a hard time keeping track of where everything was at during the beginning. Now that I have a better grasp, I can do this quick